File: src/main/java/net/moddingplayground/direbats/sound/DirebatsSoundEvents.java
package net.moddingplayground.direbats.sound;

import net.minecraft.sound.SoundEvent;
import net.minecraft.util.Identifier;
import net.minecraft.util.registry.Registry;
import net.moddingplayground.direbats.Direbats;

public class DirebatsSoundEvents {
    public static final SoundEvent ENTITY_DIREBAT_AMBIENT = direbat("ambient");
    public static final SoundEvent ENTITY_DIREBAT_HURT = direbat("hurt");
    public static final SoundEvent ENTITY_DIREBAT_ATTACK = direbat("attack");
    public static final SoundEvent ENTITY_DIREBAT_DEATH = direbat("death");
    private static SoundEvent direbat(String id) {
        return entity("direbat", id);
    }

    private static SoundEvent entity(String entity, String id) {
        return register("entity." + entity + "." + id);
    }

    private static SoundEvent register(String id) {
        Identifier identifier = new Identifier(Direbats.MOD_ID, id);
        return Registry.register(Registry.SOUND_EVENT, identifier, new SoundEvent(identifier));
    }
}

File: src/main/kotlin/dev/andante/direbats/sound/DirebatsSoundEvents.kt
package dev.andante.direbats.sound

import dev.andante.direbats.entity.DirebatsEntityTypes
import net.minecraft.entity.EntityType
import net.minecraft.registry.Registries
import net.minecraft.registry.Registry
import net.minecraft.sound.SoundEvent
import net.minecraft.util.Identifier

object DirebatsSoundEvents {
    val ENTITY_DIREBAT_AMBIENT = direbat("ambient")
    val ENTITY_DIREBAT_HURT = direbat("hurt")
    val ENTITY_DIREBAT_ATTACK = direbat("attack")
    val ENTITY_DIREBAT_DEATH = direbat("death")

    private fun direbat(id: String): SoundEvent {
        return registerEntity(DirebatsEntityTypes.DIREBAT, id)
    }

    private fun registerEntity(entity: EntityType<*>, id: String): SoundEvent {
        val identifier = Registries.ENTITY_TYPE.getId(entity)
        val soundIdentifier = Identifier(identifier.namespace, "entity.${identifier.path}.$id")
        return Registry.register(Registries.SOUND_EVENT, soundIdentifier, SoundEvent.of(soundIdentifier))
    }
}
